azurelinuxagent/common/cgroupconfigurator.py

```diff
@@ -605,7 +605,8 @@ def _check_processes_in_agent_cgroup(self):
                     current = process
                     while current != 0 and current not in agent_commands:
                         current = self._get_parent(current)
-                    if current == 0:
+                    # Process started by agent will have a marker and check if that marker found in process environment.
+                    if current == 0 and not self.__is_process_env_flag_found(process):
```

Comment:
Added as alternative condition If all the above checks unsuccessful.

Reply:
Actually, we have special cases like demon, systemd-run checks and we need those even now. I think only place we can improve when we check parent process.  We can get rid of this and add env variable check there. 
```
                    current = process
                    while current != 0 and current not in agent_commands:
                        current = self._get_parent(current)
```

Reply:
After further look I realized we need this parent check since few places like logcollector we do popen directly not going through defined code. In those scenarios, this parent pid check would catch it. So, I would say either we need all checks or update direct popen calls to shellutil. Let me know your thoughts.

Reply:
that is not quite what i meant, but we can sync offline... for now you can leave it as is